### DISCUSSION TITLE
Can't Clone a Repo

### DISCUSSION THREAD

**Discussion by inswo** [community/community#39772] [New to GitHub] [answered]
Hi everyone, 

I'm working with Windows 11.
I followed all instructions from the documentation of Github and I'm successfully authenticated.

However, when I want to clone a repo on my editor, I always end up with "git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository."

Any help would be appreciated, thanks !

**Comment by airtower-luna** [comment]
> However, when I want to clone a repo on my editor, I always end up with "[git@github.com](mailto:git@github.com): Permission denied (publickey).
> fatal: Could not read from remote repository."

This tells me two things:

1. You're using SSH to clone.
2. SSH as used by Git *cannot* authenticate to GitHub.

Obviously that leaves the question why. :sweat_smile: 

Try running `ssh -v git@github.com`. That command makes SSH (try to) connect to GitHub, and print debug information. That might provide some hints as to why authentication fails.

**Reply by inswo** [reply]
Hi @airtower-luna ,

I'm a true beginner to be honest, that might be dumb 😣

Here's the return :
debug1: Authentication succeeded (publickey).
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
You've successfully authenticated, but GitHub does not provide shell access.
Authenticated to github.com

Thanks for your help!

**Reply by airtower-luna** [reply]
Okay, that's interesting: SSH works, but Git using SSH doesn't. It's a problem I've heard about from Windows users before. I known that the problem is that Git and SSH disagree where the home directory is, and end up looking for your keys in different places, but I'm not familiar enough with Windows to say how to fix it. Maybe someone else can chip in. :sweat_smile: 

What I've seen work as a workaround is setting `GIT_SSH_COMMAND="ssh -v"`. Or rather I've told people to use that to get debug info, and then it just worked. :grin:

**Reply by inswo** [reply]
@airtower-luna , the term is not recognized, thank you though, hope somebody has the answer ahah
Which environment are you using ? 
My Mac device had no difficulty to connect 😒

**Reply by airtower-luna** [reply]
Which term do you mean? What I wrote would be an environment variable to set when calling `git`.

I've been using Linux exclusively for years, mostly (but not exclusively) Debian. :wink:

**Reply by inswo** [reply]
@airtower-luna, sorry for the late reply!
That's good to know!
I usually work on Mac OS but had to switch to Windows for better versatility.
I feel like Linus is hard to get to know with 😩

**Reply by airtower-luna** [reply]
It's a common rumor that Linux is hard to learn, but I don't think it's true. It's just different from what most people are used to. But you'll hardly find a more versatile system! :smile_cat:

**Reply by inswo** [reply]
@airtower-luna Thanks for the info!

**Comment by Preetiraj3697** [comment, marked as answer]
You use this three command:
git clone "https://github.com/your_repo.git"
cd your_repo.
git add .
git commit -m"add"
git push origin main/master.

**Reply by inswo** [reply]
@Preetiraj3697 , thanks a lot, it worked!

**Reply by Preetiraj3697** [reply]
Welcome 😊 I am happy to help you. @inswo If you don't mind please you are mark my answer then You help me to  get Galaxy Brain Badge.
